onnxruntime/test/util/test_utils.cc

```diff
@@ -117,6 +117,57 @@ void RunAndVerifyOutputsWithEP(const ORTCHAR_T* model_path, const char* log_id,
   VerifyOutputs(output_names, expected_fetches, fetches);
 }
 
+void RunAndVerifyOutputsWithEPViaModelData(std::string model_data, const char* log_id,
```

Comment:
Most of the code here can be moved into a shared function (shared with RunAndVerifyOutputsWithEP), you can pass a lambda to create inference session to the shared helper from this and RunAndVerifyOutputsWithEP,